tests/kda/test_recurrent_kda_training.py

```
@@ -217,12 +331,71 @@ def bf16(shape, multiplier=1.0):
+    names = ("q", "k", "v", "g", "beta", "A_log", "dt_bias", "initial_state")
+    args = tuple(inputs[name] for name in names)
+
+    with pytest.raises(ValueError, match="must be provided for packed tensors"):
+        recurrent_kda_training_forward(*args, inputs["cu_seqlens"])
+    with pytest.raises(ValueError, match="must be provided for packed tensors"):
+        kda_training_api._validate_forward_inputs(*args, inputs["cu_seqlens"], None)
+    with pytest.raises(ValueError, match="must be a CPU tensor"):
+        kda_training_api._validate_forward_inputs(
+            *args, inputs["cu_seqlens"], inputs["cu_seqlens"]
+        )
+    with pytest.raises(ValueError, match="dtype torch.int64"):
```

Ruff (0.16.2)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)